lib/rbac/filterer.rb

```diff
@@ -248,7 +248,7 @@ def search(options = {})
         else
           target_ids  = targets.collect(&:id)
           klass       = targets.first.class
-          klass       = base_class if !klass.respond_to?(:find) && (base_class = rbac_base_class(klass))
+          klass       = rbac_base_class(klass) unless klass.respond_to?(:find)
```

Comment:
I think the assignment is so that the value of base_class is consistent later on in case someone uses it.  With this change, if code were added later it would be the "wrong" value.  That being said, I'm probably ok with this change, but would like @gtanzillo or @lpichler to eyeball it.

Reply:
This is wrong.  This assignment is a temporal variable just so we can do a nil check on it... See https://github.com/ManageIQ/manageiq/pull/20910#pullrequestreview-561412205